debian/conkeror.1
.\" -*- nroff -*-
.TH CONKEROR 1 "June 4, 2008"
.SH NAME
conkeror \(hy a web browser modeled after Emacs
.SH SYNOPSIS
.B conkeror 
.RI [ options ] 
.RI [ url ]
.RI ...
.SH DESCRIPTION
This manual page documents briefly
.B conkeror
.PP
\fBconkeror\fP is a web browser based on xulrunner, the Mozilla
application framework that also powers Mozilla Firefox. It can be
controlled completely with the keyboard. The keybindings are very
similar to those from Emacs, so Emacs users should immediately feel at
home.
.SH OPTIONS
This program has the same options as all other xulrunner applications.
A summary of more important options is included below. All options are
listed by calling \fBconkeror \-\-help\fP.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-v, \-\-version
Show version of program.
.TP
.B \-\-display=DISPLAY
X display to use
.TP
.B \-P <profile>
Start with profile <profile>
.TP
.B \-ProfileManager
Start with ProfileManager.
.TP
.B \-no\-remote
Open new instance, not a new window in running instance.
.TP
.B \-UILocale <locale>
Start with <locale> resources as UI Locale.
.TP
.B \-safe\-mode
Disables extensions and themes for this session.
.TP
.B \-jsconsole
Open the Error console.
.SH CONCEPTS
Similar to Emacs conkeror works with windows and buffers instead of
windows and tabs.
.PP
When changing buffers with \fBC\(hyx b\fP or killing buffers with \fBC\(hyx
k\fP, you can cut down the list of buffers by typing a few characters
hich filters the list for buffers which match these characters and
then choosing them from the matching list of buffers with cursor keys.
.PP
The same idea is present with bookmarks: Bookmarks are looked up by
pressing \fBg\fP to go to a new page and then start typing parts of the
bookmarks title or URL.
.SH TIPS AND TRICKS
If you want to bookmark a page with \fBb\fP and the page has frames or
inline frames (e.g. advertisments) you will be asked which frame to
bookmark. If you want to bookmark the whole page with all frames,
press \fB0\fP.
.PP
If you want to copy the link of the current page, press \fBc\fP for
copy link and then type \fB0\fP for the current page instead of a link
on this page.
.PP
Instead of typing the link number when following or copying a link,
you can also type a part of the linked text. If don't type a number
and one links is already active (e.g. after going back one page in the
history) that link will be selected.
.SH KEYBINDINGS
.PP
When you start it up initially you see the help page with all
informations and a tutorial on how to use conkeror. You alway come
back there by typing \fBC\(hyh i\fP which means pressing and holding the
Control key (on PCs this often is the Alt key) and then pressing
"i". The tutorial can be reached by pressing \fBC\(hyh t\fP.
.PP
This is a short list of the most important keybindings:
.TP
.B C\(hyx C\(hyc
Exit the program
.TP
.B C\(hyx C\(hyf
Open an URL in a new buffer.
.TP
.B C\(hyx C\(hyv
Open an URL in the current buffer by editing the current URL.
.TP
.B C\(hyx b
Change buffer
.TP
.B C\(hyx k
Kill (close) buffer
.TP
.B M\(hyp
Change to previous buffer
.TP
.B M\(hyn
Change to next buffer
.TP
.B g
Go to another URL
.TP
.B r
Reload page
.TP
.B f
Follow a link
.TP
.B C\(hyu f
Follow a link and open it in a new window
.TP
.B c
Copy a link to clipboard
.TP
.B b
Bookmark page
.TP
.B B
Go to previous page (Backward) in page history
.TP
.B F
Go to next page (Forward) in page history
.TP
.B [[
Go to previous page (Backward) in site navigation (if present)
.TP
.B ]]
Go to next page (Forward) in site navigation (if present)
.TP
.B C\(hyg
Stop (loading a the page, incremental search, etc.)
.TP
.B C\(hys
Incremental search forward
.TP
.B C\(hyr
Incremental search backward
.TP
.B C\(hyw
Cut
.TP
.B C\(hyy
Paste
.SH KNOWN BUGS
conkeror (respectively xulrunner) sometimes hangs at \-\-help (and
possibly other situations) if /dev/random does not provide enough
entropy.

conkeror doesn't give feedback if a spawned external process crashes
or exits unsuccessfully.
.SH SEE ALSO
.BR xulrunner\-1.9(1),
.SH AUTHOR
conkeror was written by Shawn Betts, John J. Foerch, Jeremy
Maitin-Shepard and other. See /usr/share/doc/conkeror/CREDITS for a
full list of cntributors.
.PP
This manual page was written by Axel Beckert <abe@deuxchevaux.org>,
for the Debian project (but may be used by others).


ChangeLog
2007-10-25  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.39

	* chrome.manifest.for-jar: new file.

	* chrome.manifest: rewrote so conkeror can be run in-place,
	without building.

	* build.sh (do_target_xulapp): use `chrome.manifest.for-jar' when
	building the xulapp.

	* build.sh: removed unused OSX-related code.

2007-10-25  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.36

	* components/application.js (application): load save.js.

2007-10-24  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.35

	* components/application.js (application.get_os): new function.

	* components/application.js (application.set_default_directory):
	use get_os.

2007-10-22  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.32

	* test/cmdline.sh, test/batch-suite.js, test/make-temp-install.sh:
	initial checkin of test suite.

2007-10-22  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.31

	* build.sh: usage instructions say `bash build.sh <target>'

2007-10-22  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.30

	* build.sh: provide a template to mktemp.

	* install.sh: use bash build.sh xulapp.

2007-10-22  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.29

	* components/commandline.js (handle): rename performDefault to
	perform_default.

	* components/application.js (application): load keyboard.js,
	utils.js, numbering.js, bindings.js, minibuffer.js, commands.js,
	interactive.js

	* defaults/preferences/prefs.js: ui.key.generalAccessKey,
	accessibility.typeaheadfind

	* components/application.js (application.commands)
	(application.current_command): new variables.

	* components/appication.js (application): call init_webjumps and
	init_window_title.

2007-10-19  John J. Foerch  <jjfoerch@earthlink.net>

	* components/commandline.js (handle): new variable suppress_rc.
	Eval switch, `-i', changed to `-e'.  Eval switch no longer
	automatically suppresses opening a window.  Batch mode switch `-q'
	changed to `-batch'.  New switch `-l'.  New switch `-q' to
	suppress rc.  Conditionally load rc.

	* components/application.js (get_chrome_contents): use dump
	instead of dumpln.

	* components/application.js (application.set_default_directory):
	new function, and accompanying variable, `default_directory'.

	* components/application.js (application): load debug.js and
	localfile.js.

	* components/application.js (application.encode_xpcom_structure)
	(application.decode_xpcom_structure, application.add_hook)
	(application.generate_new_frame_tag)
	(application.get_frame_by_tag): new.

	* components/application.js (application.run_hooks): rewrote.  new
	parameter, `scope'.

	* components/application.js (application.generate_new_frame_name)
	(application.get_frame_by_name): removed.

	* components/application.js (application.make_frame): replaced
	`name' with `tag'.  new way of encoding arguments to pass to the
	frame.  run make_frame_hook.

2007-10-17  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.27

	* build.sh: removed references to install.rdf.

	* build.sh: preprocess application.js instead of conkeror.js.

	* components/application.js (application): new field, `version'.

	* components/application.js (application): new field, `start_time'.

	* components/application.js (application): new field, `preferences'.

2007-10-17  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.26

	* htmlBindings.xml: removed file.

	* patformHTMLBindings.xml: removed file.

	* install.js: removed file.

	* install.rdf: removed file.

2007-08-23  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.23

	* components/commandline.js: extensive changes.  much code moved
	to application.js.

	* components/application.js: new file.

	* build.sh (copy_tree_sans_boring, do_target_jar): rewritten find
	calls.

	* conkeror/locale/en-US/gui_context_menu.properties:
	context_copy_frameset_frame_location.label and
	context_copy_frameset_frame_location.accesskey new.

	* defaults/preferences/prefs.js: dom.disable_open_during_load,
	javascript.options.showInConsole,
	dom.disable_window_open_feature.status,
	dom.disable_window_open_feature.location,
	dom.disable_window_status_change, dom.disable_window_move_resize,
	dom.disable_window_flip new.

2007-05-20  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.16

	* defaults/preferences/find.js: new file.

	* conkeror/locale/en-US/gui_context_menu.properties: new file.

	* chrome.manifest: added locale conkeror en-US.

2007-05-15  John J. Foerch  <jjfoerch@earthlink.net>

	* VERSION: 0.42.15

	* components/commandline.js (conkeror): new global.

	* components/commandline.js (init_conkeror): new function.

	* components/commandline.js (myAppHandler): renamed to
	conkerorHandler.

	* components/commandline.js (conkerorHandler.handle): rewritten.

	* components/commandline.js (myAppHandlerModule): renamed to
	conkerorModule.

2007-05-10  John J. Foerch  <jjfoerch@earthlink.net>

	* defaults/preferences/prefs.js: browser.chromeURL

2007-05-10  John J. Foerch  <jjfoerch@earthlink.net>

	* components/nsCrank.js: removed.

	* conkeror/branding/brand.dtd: new file.

	* conkeror/branding/brand.properties: new file.

	* application.ini (XRE): EnableExtensionManager=1.

	* chrome.manifest: branding entry.

	* default/preferences/prefs.js: removed defaultChromeURI.

	* components/commandline.js (jsdump, consoledump, consoledumpln):
	new.



modules/conkeror.js
/**
 * (C) Copyright 2007-2008 Jeremy Maitin-Shepard
 *
 * Use, modification, and distribution are subject to the terms specified in the
 * COPYING file.
**/

var start_time = Date.now ();

require("keywords.js"); // not required to be listed as dep.
require("labels.js"); // not required to be listed as a dep.
require("coroutine.js"); // not required to be listed as dep.

require("hook.js");
require("debug.js");
require("utils.js");
require("interactive.js");
require("minibuffer.js"); // depends: interactive.js
require("minibuffer-read.js");
require("minibuffer-read-option.js");
require("minibuffer-completion.js");
require("minibuffer-read-file.js");
require("timer.js");
require("spawn-process.js");
require("mime.js");
require("keyboard.js");
require("buffer.js");
require("window.js");
require("download-manager.js");

require("element.js");

require("universal-argument.js");
require("commands.js"); // depends: interactive.js
require("webjump.js"); // depends: interactive.js
require("history.js");

require("save.js");

require("zoom.js");
require("follow-relationship.js");

require("find.js");

require("hints.js");

require("help.js");

require("rc.js");

require("ssl.js");

require("media.js");

require("command-line.js");

require("search-engine.js");

require("permission-manager.js");
require("cookie.js");
require("cache.js");

require("theme.js");
